tests/manual/monkey/src/com/oracle/tools/fx/monkey/MonkeyTesterApp.java

```diff
@@ -0,0 +1,247 @@
+            updatePage(c);
+        });
+
+        contentPane = new BorderPane();
+        contentPane.setId("contentPane");
+
+        SplitPane split = new SplitPane(pageSelector, contentPane);
+        split.setDividerPositions(0.15);
+        SplitPane.setResizableWithParent(pageSelector, Boolean.FALSE);
+        SplitPane.setResizableWithParent(contentPane, Boolean.TRUE);
+
+        BorderPane bp = new BorderPane();
+        bp.setTop(createMenu());
+        bp.setCenter(split);
+        bp.setBottom(st);
+
+        stage.setScene(new Scene(bp));
+        stage.setWidth(1200);
+        stage.setHeight(800);
+
+        stage.renderScaleXProperty().addListener((x) -> updateStatus());
+        stage.renderScaleYProperty().addListener((x) -> updateStatus());
+        updateTitle();
+        updateStatus();
+
+        stage.show();
+    }
+
+    protected MenuBar createMenu() {
+        CheckMenuItem orientation = new CheckMenuItem("Orientation: RTL");
+        orientation.setOnAction((ev) -> {
+            NodeOrientation v = (orientation.isSelected()) ? NodeOrientation.RIGHT_TO_LEFT
+                : NodeOrientation.LEFT_TO_RIGHT;
+            stage.getScene().setNodeOrientation(v);
+        });
+
+        MenuBar b = new MenuBar();
+        // File
+        FX.menu(b, "_File");
+        FX.item(b, "Quit", Platform::exit);
+        // Page
+        FX.menu(b, "_Page");
+        FX.item(b, "Reload Current Page", this::reloadCurrentPage);
+        // Menu
+        FX.menu(b, "_Menu");
+        ToggleGroup g = new ToggleGroup();
+        FX.radio(b, "RadioMenuItem 1", KeyCombination.keyCombination("Shortcut+1"), g);
+        FX.radio(b, "RadioMenuItem 2", KeyCombination.keyCombination("Shortcut+2"), g);
+        FX.radio(b, "RadioMenuItem 3", KeyCombination.keyCombination("Shortcut+3"), g);
+        FX.menu(b, "_Tools");
+        FX.item(b, "Native-to-ascii", this::openNative2Ascii);
+        // Window
+        FX.menu(b, "_Window");
+        FX.item(b, orientation);
+        FX.separator(b);
+        FX.item(b, "Open Modal Window", this::openModalWindow);
+        return b;
+    }
+
+    protected void updatePage(DemoPage p) {
+        FxSettings.store(contentPane);
+        currentPage = p;
+        contentPane.setCenter(p == null ? null : p.createPane());
+        updateTitle();
+        FxSettings.restore(contentPane);
+    }
+
+    protected void reloadCurrentPage() {
+        updatePage(currentPage);
+    }
+
+    protected void updateTitle() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("Monkey Tester");
+        if (currentPage != null) {
+            sb.append(" - ");
+            sb.append(currentPage.toString());
+        }
+        stage.setTitle(sb.toString());
+    }
+
+    protected void updateStatus() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("   ");
+        sb.append(System.getProperty("javafx.runtime.version"));
```

Comment:
I would love to see more information in the status bar - especially if users will submit screenshots as a proof of the fact for bugs having additional information would be super nice. Already love the scale output, would love to see the current window size in either logical or physical pixels and a version of JVM in addition to the version of the JFX runtime - i think it might be even more important if we would add the JFX/Swing interoperability tests to this tool. If not in a status bar may be some sort of about menu that reports all the details about the environment such as reported OS version, number of displays, reported graphics mode with insets and such. And i think that might be an RFE for later version of the tool.

Reply:
thank you for suggestions!  added to https://bugs.openjdk.org/browse/JDK-8307176